node/src/integration-test/kotlin/net/corda/node/jmx/Publish.kt

```diff
@@ -0,0 +1,29 @@
+package net.corda.node.jmx
+
+import net.corda.testing.driver.DriverParameters
+import net.corda.testing.driver.JmxPolicy
+import net.corda.testing.driver.driver
+import org.junit.Test
+import java.net.HttpURLConnection
+import java.net.URL
+import kotlin.test.assertTrue
+
+class Publish {
+
+    @Test(timeout=300_000)
+    fun `node publishes node information via JMX when configured to do so`() {
+        driver(DriverParameters(notarySpecs = emptyList(), jmxPolicy = JmxPolicy.defaultEnabled())) {
+            val jmxAddress = startNode().get().jmxAddress.toString()
+            val nodeStatusURL = URL("http://$jmxAddress/jolokia/read/net.corda:*")
+            var httpResponse = HttpURLConnection.HTTP_NOT_FOUND
```

Comment:
I am trying not to use `var` unless I absolutely have to. In this case, I think this should work:
```suggestion
                       val httpResponse = with(nodeStatusURL.openConnection() as HttpURLConnection) {
                requestMethod = "GET"
                responseCode
            }
```